org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/viewer/MavenTextEditingSupport.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Patrick Ziegler and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Patrick Ziegler - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.m2e.pde.ui.target.editor.internal.viewer;
+
+import org.eclipse.jface.viewers.CellEditor;
+import org.eclipse.jface.viewers.ColumnViewer;
+import org.eclipse.jface.viewers.EditingSupport;
+import org.eclipse.jface.viewers.TextCellEditor;
+import org.eclipse.m2e.pde.target.MavenTargetDependency;
+import org.eclipse.m2e.pde.ui.target.editor.internal.MavenAccessor;
+import org.eclipse.swt.widgets.Composite;
+
+/**
+ * This class provides the ability to edit the text fields of Maven
+ * dependencies. Supported fields are:
+ * <ul>
+ * <li>group id</li>
+ * <li>artifact id</li>
+ * <li>version</li>
+ * <li>classifier</li>
+ * </ul>
+ */
+public final class MavenTextEditingSupport extends EditingSupport {
+	private final MavenAccessor accessor;
+
+	public MavenTextEditingSupport(ColumnViewer viewer, MavenAccessor accessor) {
```

Comment:
Probably it would be better instead of having `MavenAccessor`  to pass the getter/setter here to remove an indirection?

Reply:
Having this indirection was the initial idea. In the end, I removed this class completely in favor of directly passing on the getter/setter.